Core/Core/Offline/Model/OfflineFileInteractor.swift

```diff
@@ -21,19 +21,42 @@ import Combine
 public protocol OfflineFileInteractor {
     func filePath(sessionID: String, courseId: String, fileID: String, fileName: String) -> String
     func isItemAvailableOffline(courseID: String?, fileID: String?) -> Bool
+    func filePath(source: OfflineFileSource?) -> String?
+    func isItemAvailableOffline(source: OfflineFileSource?) -> Bool
     var isOffline: Bool { get }
 }
 
 public final class OfflineFileInteractorLive: OfflineFileInteractor {
-
-    // MARK: - Dependencies
-
     private let offlineModeInteractor: OfflineModeInteractor
 
     public init(offlineModeInteractor: OfflineModeInteractor = OfflineModeAssembly.make()) {
         self.offlineModeInteractor = offlineModeInteractor
     }
 
+    // MARK: Public functions
+
+    public func filePath(source: OfflineFileSource?) -> String? {
+        switch source {
+        case .privateFile(let sessionID, let courseID, let sectionName, let resourceID, let fileID):
+            return filePath(sessionID: sessionID, courseId: courseID, section: sectionName, resourceId: resourceID, fileID: fileID)
+        case .publicFile(let sessionID, let courseID, let fileID, let fileName):
+            return filePath(sessionID: sessionID, courseId: courseID, fileID: fileID, fileName: fileName)
+        case .none:
+            return nil
+        }
+    }
+
+    public func isItemAvailableOffline(source: OfflineFileSource?) -> Bool {
+        switch source {
+        case .privateFile(let sessionID, let courseID, let sectionName, let resourceID, let fileID):
+            return isItemAvailableOffline(sessionID: sessionID, courseId: courseID, section: sectionName, resourceId: resourceID, fileID: fileID)
+        case .publicFile(let sessionID, let courseID, let fileID, let fileName):
+            return isItemAvailableOffline(courseID: courseID, fileID: fileID)
```

Comment:
I think this might not work in all cases as `isItemAvailableOffline` checks `offlineSyncSelections`, but hidden files won't be saved to that selection.